pkg/node/drive_sync.go

```diff
@@ -0,0 +1,159 @@
+// This file is part of MinIO DirectPV
+// Copyright (c) 2021, 2022 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package node
+
+import (
+	"context"
+	"strings"
+
+	"github.com/google/uuid"
+	directcsi "github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta3"
+	"github.com/minio/directpv/pkg/client"
+	"github.com/minio/directpv/pkg/sys"
+	"github.com/minio/directpv/pkg/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+)
+
+func syncDrives(ctx context.Context, nodeID string, loopbackOnly bool, topology map[string]string) error {
+	devices, err := sys.ProbeDevices()
+	if err != nil {
+		klog.ErrorS(err, "unable to probe devices")
+		return err
+	}
+
+	resultCh, err := client.ListDrives(
+		ctx,
+		[]utils.LabelValue{utils.NewLabelValue(nodeID)},
+		nil,
+		nil,
+		client.MaxThreadCount,
+	)
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+
+	for result := range resultCh {
```

Comment:
This is wrong. I wanted to get all the drives first before trying to match. By doing that, we ensure we can do elimination logic for matching.